src/platform_impl/linux/x11/util/geometry.rs

```diff
@@ -156,8 +156,34 @@ impl XConnection {
                 root,
                 0,
                 0,
-                &mut coords.x_rel_root,
-                &mut coords.y_rel_root,
+                &mut coords.x_rel,
+                &mut coords.y_rel,
+                &mut coords.child,
+            );
+        }
+
+        self.check_errors()?;
+        Ok(coords)
+    }
+
+    pub fn translate_coords(
+        &self,
+        src_w: ffi::Window,
+        dest_w: ffi::Window,
+        src_x: i32,
+        src_y: i32,
+    ) -> Result<TranslatedCoords, XError> {
+        let mut coords = TranslatedCoords::default();
+
+        unsafe {
+            (self.xlib.XTranslateCoordinates)(
+                self.display,
+                src_w,
+                dest_w,
+                src_x,
+                src_y,
+                &mut coords.x_rel,
+                &mut coords.y_rel,
                 &mut coords.child,
```

Comment:
I'm not entirely sure why you need a second method here, given that the only difference is the `(0, 0)`? Though maybe it's fine to have it that way.

Reply:
the original method was mapping window -> root while I need the opposite (root -> window)

Reply:
I could keep the new one (because it is more flexible) and remove the old one and migrate its usages to the new one.

Reply:
I mean that you path both Root and other window, so you can swap them? Would be the other way around.

Reply:
Sorry, I am not sure I understand what you mean.

Reply:
I mean that your new call supersedes the existing one, since with it you can express both translations just fine. The old one wasn't working due to inability to pass `src_x` and `src_y` and hardcoded it to zero.

But I guess it's fine the way it is since you don't have remember that you need `(0, 0)` for one of the translations.